tests/test_auto_reload.py

```diff
@@ -0,0 +1,85 @@
+import os
+import sys
+import subprocess
+import signal
+from threading import Thread
+import requests
+from time import sleep
+
+sanic_project_content_one  = '''
+from sanic import Sanic
+from sanic import response
+
+app = Sanic(__name__)
+
+
+@app.route("/")
+async def test(request):
+    return response.json({"test": 1})
+
+
+if __name__ == '__main__':
+    app.run(host="127.0.0.1", port=8000, auto_reload=True)
+'''
+
+sanic_project_content_two  = '''
+from sanic import Sanic
+from sanic import response
+
+app = Sanic(__name__)
+
+
+@app.route("/")
+async def test(request):
+    return response.json({"test": 2})
+
+
+if __name__ == '__main__':
+    app.run(host="127.0.0.1", port=8000, auto_reload=True)
+'''
+
+process_id = None
+def execute_cmd(command):
+    process = subprocess.Popen(command, shell=True)
+    global process_id
+    process_id = process.pid
+    process.communicate()
+
+
+class TestAutoReloading:
+
+    def test_reloading_after_change_file(self,capsys):
+        if os.name != 'posix':
+            return
+
+        with capsys.disabled(): pass
+        sanic_app_file_path = "simple_sanic_app.py"
+        with open (sanic_app_file_path, "w") as _file:
+            _file.write(sanic_project_content_one)
+
+        cmd = ' '.join([sys.executable, sanic_app_file_path])
+        thread = Thread(target=execute_cmd, args=(cmd,))
+        thread.start()
+
+        sleep(2) # wait for completing server start process
+        response = requests.get("http://127.0.0.1:8000/").json()
+        assert response == {"test": 1}
+
+        with open (sanic_app_file_path, "w") as _file:
+            _file.write(sanic_project_content_two)
+        sleep(2) # wait for completing server start process
+        response = requests.get("http://127.0.0.1:8000/").json()
+        assert response == {"test": 2}
+
+        thread.join(1)
+        os.remove(sanic_app_file_path)
+
+    def teardown_method(self, method):
+            if process_id:
```

Comment:
Seems like the spacing is off here.

Reply:
@seemethere 
Thanks for your review :)
I will fix this problems as soon as I can, but there is one point:
I'm using Linux and actually I can't write programs for windows. Now this mechanism works on Linux  and I added an condition to check platform.
If somebody add windows support to it, it will be great :)

Reply:
No problem! Thank you for your contribution!, if you could add a quick failure, raising a `NotImplementedError` should do the trick, for those on windows until we can have someone on windows verify that this method works.

Reply:
I will work on it in this weekend!

Reply:
Done!